components/core/tests/test-ffi_KeyValuePairLogEvent.cpp

```diff
@@ -274,63 +313,90 @@ TEST_CASE("ffi_KeyValuePairLogEvent_create", "[ffi]") {
             {4, "d", SchemaTree::Node::Type::Str},
             {4, "d", SchemaTree::Node::Type::Float},
             {3, "e", SchemaTree::Node::Type::Obj},
-            {4, "f", SchemaTree::Node::Type::Obj}
+            {4, "f", SchemaTree::Node::Type::Obj},
+            {SchemaTree::cRootId, "a", SchemaTree::Node::Type::Int},
+            {1, "b", SchemaTree::Node::Type::Bool}
     };
     for (auto const& locator : locators) {
-        REQUIRE_NOTHROW(schema_tree->insert_node(locator));
+        REQUIRE_NOTHROW(auto_generated_schema_tree->insert_node(locator));
+        REQUIRE_NOTHROW(user_generated_schema_tree->insert_node(locator));
     }
 
+    // This test case implicitly requires the auto-generated and user-generated schema trees to
+    // be identical. Adding this check to prevent this assumption is broken by future development.
```

Comment:
Can you explain this comment? I think it makes sense that we're checking that the two schema trees are identical considering we inserted the same nodes into them. So this comment is a bit confusing to me since it makes it sound like that's not what the code is supposed to do?

Reply:
This is what the code is supposed to do. It's added as a hint to tell future developers don't break this assumption, so that if the assertion fails they know what need to be fixed

Reply:
But if the assertion fails, they should look at the test and realize why it failed, right? Like I think this comment is necessary if the test is complicated, but this one seems very simple to read, so the comment makes it seem like it's doing something more complicated than it is.